erpnext/buying/doctype/purchase_order/purchase_order.js

```diff
@@ -428,7 +428,8 @@ erpnext.buying.PurchaseOrderController = class PurchaseOrderController extends e
 					},
 					allow_child_item_selection: true,
 					child_fielname: "items",
-					child_columns: ["item_code", "qty"]
+					child_columns: ["item_code", "qty"],
```

Comment:
`item_code` can appear twice because we allow repeating same item. Here row's `name` needs to be checked.

Reply:
We can add the same item more than once but cannot add an item twice from the same Material Request.
<img width="1095" alt="Screenshot 2021-11-29 at 5 24 48 PM" src="https://user-images.githubusercontent.com/63660334/143864242-a73ecc39-06d3-47b9-86d3-f9173f59bbb1.png">

Reply:
That's not what I meant. A material request can have multiple rows for same item code, so we shouldn't be checking for duplicate using `item_code`. Use `material_request_item` which is hash of row. 

<img width="1440" alt="Screenshot 2021-11-30 at 6 33 55 PM" src="https://user-images.githubusercontent.com/9079960/144052388-ce30d1a3-7409-4d79-b1ea-355e1549dae8.png">